cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,248 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Pytest test configuration file.
+
+"""
+import pytest
+from pathlib import Path
+from subprocess import run, TimeoutExpired
+from time import sleep
+import json
+import os
+import shutil
+
+@pytest.fixture(scope="session")
+def installer_path(request):
+    return Path(request.config.getoption("--installer-path")).resolve()
+
+@pytest.fixture(scope="session")
+def install_root(request):
+    return Path(request.config.getoption("--install-root")).resolve()
+
+@pytest.fixture(scope="session")
+def project_path(request):
+    # MSBuild doesn't want us to output in the temp directory MSB8029
+    # so we use a pre-defined folder outside of %TMP%
+    #return tmpdir_factory.mktemp("TestProject")
+    path = Path(request.config.getoption("--project-path")).resolve()
+    yield path
+
+    # wait a few seconds for processes to quit
+    sleep(5)
+
+    shutil.rmtree(path)
+
+@pytest.fixture(scope="session")
+def cmake_path(install_root):
+    return install_root / 'cmake' / 'runtime' / 'cmake-3.20.2-windows-x86_64' / 'bin' / 'cmake.exe'
+
+@pytest.fixture(scope="session")
+def test_installer(installer_path, install_root):
+    assert installer_path.is_file(), f"Invalid installer path {installer_path}"
+
+    # when the installer is run with timeout of 30 minutes
+    result = run([installer_path , f"InstallFolder={install_root}", "/quiet"], timeout=30*60)
+
+    # the installer succeeds
+    assert result.returncode == 0, f"Installer failed with exit code {result.returncode}"
+
+    # the install root is created
+    assert install_root.is_dir(), f"Invalid install root {install_root}"
+
+@pytest.mark.parametrize(
+    "filename", [
+        pytest.param('Editor.exe'),
+        pytest.param('AssetProcessor.exe'),
+        pytest.param('MaterialEditor.exe'),
+        pytest.param('o3de.exe'),
+    ]
+)
+def test_binaries_exist(test_installer, install_root, filename):
+    bin_folder = install_root / 'bin/Windows/profile/Default'
+    assert (bin_folder / filename).is_file(), f"{filename} not found in {bin_folder}"
+
+
+@pytest.fixture(scope="session")
+def test_o3de_registers_engine_fixture(install_root, test_installer):
+    try:
+        bin_folder = install_root / 'bin' / 'Windows' / 'profile' / 'Default'
```

Comment:
Probably worth creating a helper function to getting the bin path since this is used in a number of places